app/domain/operations/transmittable/create_job.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+module Operations
+  module Transmittable
+    # create job operation that takes params of key (required), started_at(required), publish_on(required)
+    class CreateJob
+      include Dry::Monads[:result, :do, :try]
+
+      def call(params)
+        values = yield validate_params(params)
+        job_hash = yield create_job_hash(values)
+        job_entity = yield create_job_entity(job_hash)
+        job = yield create_job(job_entity)
+        Success(job)
+      end
+
+      private
+
+      def validate_params(params)
+        return Failure('Cannot create a job without a key as a symbol') unless params[:key].is_a?(Symbol)
+        return Failure('Cannot create a job without a started_at as a DateTime') unless params[:started_at].is_a?(DateTime)
+        return Failure('Cannot create a job without a publish_on as a DateTime') unless params[:publish_on].is_a?(DateTime)
+
+        Success(params)
+      end
+
+      def create_job_hash(values)
+        Success({
```

Comment:
we should potentially add a rescue on this that would return a failure in case any of the sub methods fail.